src/node_config_file.cc

```diff
@@ -52,106 +163,71 @@ ParseResult ConfigReader::ParseNodeOptions(
     std::string prefix = "--";
     auto it = env_options_map.find(prefix.append(key));
     if (it != env_options_map.end()) {
-      switch (it->second) {
-        case options_parser::OptionType::kBoolean: {
-          bool result;
-          if (ondemand_value.get_bool().get(result)) {
-            FPrintF(stderr, "Invalid value for %s\n", it->first.c_str());
-            return ParseResult::InvalidContent;
-          }
+      // If the option has already been set, return an error
+      if (unique_options_.contains(it->first)) {
+        FPrintF(stderr, "Option %s is already defined\n", it->first.c_str());
+        return ParseResult::InvalidContent;
+      }
+      ParseResult result = ProcessOptionValue(key,
+                                              it->first,
+                                              ondemand_value,
+                                              it->second,
+                                              &node_options_,
+                                              &unique_options_);
```

Comment:
we can move unique_options outside the `ProcessOptionValue`

Reply:
done